### ISSUE TITLE
`prefer-default-parameters` rule is broken in some cases

### ISSUE THREAD

**Issue by qeleb** [sindresorhus/eslint-plugin-unicorn#3628]
# `unicorn/prefer-default-parameters` rule doesn't provide editor suggestions in some cases.

## Description
The rule doesn't suggest fixes when equivalent syntax, `||=` or `??=` are used.

With the `logical-assignment-operators` rule, cases that would yield editor suggestions are auto-fixed into cases that do not:
1.  `foo = foo || 'bar'` ✔️ ➡️ `foo ||= 'bar'` ✖️
1.  `foo = foo ?? 'bar'` ✔️ ➡️ `foo ??= 'bar'` ✖️

## Examples
1. This usage is not detected as incorrect
   ```js
   function abc(foo) {
     foo ||= 'bar';
   }
   ```

   It should have a suggested fix to be:
   ```js
   function abc(foo = 'bar') {
   }
   ```

2. The same is true for optional chaining
   ```js
   function abc(foo) {
     foo ??= 'bar';
   }
   ```

   It should have a suggested fix to be:
   ```js
   function abc(foo = 'bar') {
   }
   ```
